_posts/2016-10-06-database-testing-with-tsqlt.md

```diff
@@ -0,0 +1,87 @@
+In order to have flexibility and to allow changes in the database to be made safely; you need to have a test suite to run and check for regressions. In this post we will have a look at a testing framework for Microsoft SQL Server databases.
+
+
+##Environment
+
+###Installing MS SQL Server
+
+Get started installing [SQL Server Express](https://www.microsoft.com/en-us/download/details.aspx?id=52679)
+.
+SQL Server Express is a free lightweight version of Microsoft’s SQL Server database.
+
+###Installing MSSQL Server Manager
+
+SQL Server Management studio provides us with a powerful user interface to access, manage and configure our MS SQL database server.
+If you don’t already have it, [install it](https://msdn.microsoft.com/en-us/library/mt238290.aspx). Please note this will take some time.
+
+###Downloading tSQLt
+
+We are going to use the [tSQLt](http://tsqlt.org/) unit testing framework to test our Microsoft SQL Server database.
+We first need to get the [download](http://tsqlt.org/downloads/) from the official website. 
+
+The tSQLt [open source](https://github.com/tSQLt-org/tSQLt) framework allows us to run our tests within transactions which makes our tests independant and takes care of cleaning up after it runs. The framework also gives us the ability to isolate our tests through fake tables and stored procedure (SP) spies.
+
+##Running the test sample
+
+Once you have SQL Server Express up and running, and MSSQL Management Studio installed, open the latter and connect to your server.
+
+{% img /assets/img/custom/blog/db-testing/db-testing-1.png "" %}
+
+The tSQLt team provides a quick start example database and a set of tests to help developers get started quickly on how the framework works.
+Let’s try the example together.
+First unzip the tSQLt folder you downloaded, and make sure that CLRs are enabled on your development server. Enabling [CLR integration](https://msdn.microsoft.com/en-us/library/ms254498v=vs.110.aspx) allows us to run managed code, such as C#, on our SQL Server database.
+
+{% img /assets/img/custom/blog/db-testing/db-testing-2.png "" %}
+
+Open the Example.sql file that you will find within the unzipped folder, and execute it in your database server. This will create a test database named `tSQLt_Example`.
+
+Once your database is created, open a new query window on that DB and run the following command:
+`EXEC tSQLt.RunAll`
+This is a command to run all the tests. You should see a failing test in the result screen:
+
+{% img /assets/img/custom/blog/db-testing/db-testing-3.png "" %}
+
+Test case number 11, named test ready for experimentation if 2 particles, is failing. Let’s open the file and see what’s wrong with it; we will need to navigate to the stored procedures folder to find the test.

```

Comment:
Add `"` or  `` ` `` quotes to the test name to separate it from the rest of the sentence.